Customizable upgrade

tvdeyen opened the pull request:
Alchemy upgrader runs several rake tasks in a specific order.
This is sometimes not what you want or even breaks upgrades.
In order to customize the upgrade process you can now run each of tasks seperately.

pascalj commented:
I think it would be better to split up the rake task some more and document that.

Either we document the current tasks that are executed by `alchemy:upgrade` (and in which order) or we introduce aliases like `alchemy:upgrade:install_migrations`. That way the user can decide which task they want to execute in which order.

mamhoff commented:
I agree with @pascalj here; I would like to see the upgrader more like a set of interdependent rake tasks rather than one class that runs things. If at all possible, I think this is an easier to understand path than setting environment variables.

tvdeyen commented:
I updated the PR @mamhoff @pascalj 

I think we need to distinguish between upgrade preparation and individual upgrade steps (I name this steps on purpose because the aren't Rake tasks):

What I want to achieve here is a customizable upgrade preparation. Former we ran `alchemy:install:migrations`, `db:migrate`, `Alchemy::Seeder.seed!` in one `setup` called method in the `Alchemy::Upgrader` class. This is sometimes not ideal, so that's why I split this up into individual  Rake tasks.

Regarding the individual upgrade steps:

Yes, I would be ideal to have each upgrade step in individual rake tasks, but this is much more work to be done. Also, I don't like having too much code in a (untested) Rake task. So, that's why we took the approach to have our own task runner with individual classes representing each upgrade step. 

We could (and should) convert these custom tasks in rake tasks, invoking these classes. But this, is another PR

tvdeyen commented:
@mamhoff @pascalj added another PR (#1154)  that splits up all upgrade steps into rake tasks
